### ISSUE TITLE
Struct to_json doesn't work properly with Rails 4.1

### ISSUE THREAD

**Issue by gorkunov** [ohler55/oj#219]
```ruby
o = Struct.new(:foo).new('bar')
Oj.dump(o)
=> "\"#<struct foo=\\\"aa\\\">\""
Oj.dump(o.as_json)
=> "{\"foo\":\"aa\"}"

oj (2.11.4)
oj_mimic_json (1.0.1)
rails (4.1.8)

**Comment by ohler55** [comment]
I'll take a look at it tonight.

**Comment by ohler55** [comment]
I have a release candidate checked into github. I have not finished testing on all ruby version yet though.

**Comment by ohler55** [comment]
Released 2.11.5 with the fixes and associated unit tests. No response so closing the issue.

**Comment by gorkunov** [comment]
Works for me now. Thanks!
